### ISSUE TITLE
feature request: desktop switcher should show an outline of windows instead of just a number/name

### ISSUE THREAD

**Issue by bzb2** [lxqt/lxqt#1767]
Lubuntu 19.10 
Linux 5.3.0-40-generic #32-Ubuntu SMP

In Lubuntu 18.xx with lxde, the desktop switcher widget shows an outline of windows currently on each desktop. In lxqt, the desktop switcher widget only show a label for each desktop button. It would be very informative if this desktop switcher can show the same embedded window outlines

My system have the following items installed...
https://launchpadlibrarian.net/467546069/Dependencies.txt

**Comment by tsujan** [comment]
LXDE relied on X11+Openbox, while LXQt doesn't: it can be used with any DE-agnostic WM and should be made ready for Wayland at some point. I'm not sure it's possible.

**Comment by wxl** [comment]
I suspect this is consistent with #668 where ultimately the issue is that the window manager should be responsible for taking care of workspaces. I'm not sure it's reasonable to assume that all window managers are going to be able to provide the same information.

**Comment by tsujan** [comment]
@wxl 
We use `KWindowSystem` and so, we can do what `KWindowSystem` allows us to do; nothing more, nothing less. Particularly, no one should add X11-specific codes to LXQt.

Why `KWindowSystem`? Because it provides methods that Qt doesn't and, with it, we don't need to worry about X11 codes.

**Comment by wxl** [comment]
@tsujan Then maybe as a nice wishlist thing we could make use of that. 

@bzb2 Another way to read this discussion is that there's no hope of change in Lubuntu with the current window manager. Openbox was long ago named a "finished" project so it sees no updates. Maybe one day we'll fix that or just do the more wise thing and switch to the likes of KWin, which you could very well do in your Lubuntu system (I don't use Openbox personally but prefer awesomewm). In the meantime, what you see is what you get.

**Comment by tsujan** [comment]
@wxl I didn't close this issue because the suggested idea is good. It'll be nice if we can not only show outlines but also move windows between virtual desktops by DNDing their outlines -- of course, provided that the desktop switcher is big enough (that depends on user settings).

When Wayland becomes more mature, we might need to use KWin, as LXDE used Openbox. I'm not sure about that at all (actually, I don't like it -- I prefer LXQt to remain agnostic to WMs) but, *if it happens*, extra features could be added.

**Comment by wxl** [comment]
@tsujan I agree it's a good idea if we can make it happen. I also agree it would be best to remain as agnostic as possible with regards to external components, window managers included.

**Comment by tsujan** [comment]
In case some users don't know it: Many extra features are already available to LXQt if KWin is used with it. The same is true with older WMs, like Compiz-reloaded and xfwm4. That's the beauty of being WM-agnostic. An example with KWin (Alt+Tab):

![extra_features](https://user-images.githubusercontent.com/981076/75826863-b11ea880-5dbd-11ea-8167-7cec69ab6a0d.png)

**Comment by ThomasD13** [comment]
I switched from Lubuntu 18.04 to 22.04, and I am now also a user of Lqxt. 
I would kindly ask about the status of this feature? It was very convenient for me, to see in the taskbar which applications are placed on which virtual desktop.

I don't understand really why this feature isn't available anymore. Is there any 3 third party software, which enables this for Openbox?

Best Regards

**Comment by pheek** [comment]
I also changed from lxde to lxqt. But seeing no benefit in lxqt, I think I switch back. The best feature was, that in lxde I can move windows from one desktop to another, by draging the window-symbol (rectangle) from one place in the desktop-pager to another desktop (in the pager).

What is the task (expend) to implement this in lxqt? As a c/c++/Java/Python/Javascript Programmer I could be able to implement it, but I need to know which APIs and IDEs I should use for this.

Is anyone working on this?

**Comment by stefonarch** [comment]
I'm  not  sure but this feature in LXDE is probably tied to openbox, in LXQt being WM-agnostic this could not be possible eventually, bt I see that `wmctrl` has the option to move windows to desktop though.

>Is anyone working on this?

No, but we accept PRs

>changed from lxde to lxqt. But seeing no benefit in lxqt

There are many improvements you probably didn't noticed yet ;)
